relengapi/blueprints/archiver/__init__.py

```
@@ -100,12 +158,30 @@ def get_archive(src_url, key, preferred_region):
     region = preferred_region if preferred_region and preferred_region in buckets else random_region
     bucket = buckets[region]
     s3 = current_app.aws.connect_to('s3', region)
+    session = current_app.db.session('relengapi')
 
     # first, see if the key exists
     if not s3.get_bucket(bucket).get_key(key):
         task_id = key.replace('/', '_')  # keep things simple and avoid slashes in task url
-        if create_and_upload_archive.AsyncResult(task_id).state != 'STARTED':
-            # task is currently not in progress so start one.
+        # can't use unique support:
+        # api.pub.build.mozilla.org/docs/development/databases/#unique-row-support-get-or-create
+        # because we want to know when the table doesn't exist before creating it
+        tracker = tables.ArchiverTask.query.filter(tables.ArchiverTask.task_id == task_id).first()
+        if tracker and tracker.state in FINISHED_STATES:
+            log.info('Task tracker: {} exists but finished with state: '
+                     '{}'.format(task_id, tracker.state))
+            # remove tracker and try celery task again
+            delete_tracker(tracker)
+            tracker = None
+        if not tracker:
+            log.info("Creating new celery task and task tracker for: {}".format(task_id))
+            try:
+                session.add(tables.ArchiverTask(task_id=task_id, created_at=now(),
+                                                src_url=src_url, state="PENDING"))
+                session.commit()
+            except sa.exc.IntegrityError:
+                log.error("Could not add tracker {} to ArchiveTask table".format(task_id))
+                session.rollback()
```

I've created a new test to cover lines in this block here: https://coveralls.io/builds/3060791/source?filename=relengapi%2Fblueprints%2Farchiver%2F__init__.py#L171

I also removed the catch when adding or deleting tracking records. In either case, it would be better to bubble up the error and fail the request. I left the IntegrityError catch on update as rolling back  and continuing wouldn't be the end of the world. the tracker state is more of a debugging tool as we rely of the AsyncResult.state for flow behaviour.